src/mount_media.c

```diff
@@ -76,13 +78,110 @@ check_volsetid(const char *volid)
 	return (ret);
 }
 
+/*
+ * Mounts and tests supplied path to iso file for supplied volid
+ * Returns 0 if volid found, -1 otherwise.
+ */
+static int
+test_iso_file(const char *path, const char *volid)
+{
+	int ret = -1;
+	char opts[MAX_MNTOPT_STR];
+
+	strcpy(opts, HSFS_OPTS);
+	if (verbose)
+		printf("%s: mount hsfs (iso)\n", path);
+	if (mount(path, "/.cdrom", MS_RDONLY | MS_OPTIONSTR,
+		"hsfs", NULL, 0, opts, sizeof (opts)) != 0)
+		return (ret); /* mount failed */
+
+	/* Mounted, see if it's the image we're looking for, unmount if not */
+	ret = check_volsetid(volid);
+	if (ret != 0) {
+		if (verbose)
+			printf("%s: wrong ID, unmounting\n", path);
+		(void) umount("/.cdrom");
+	}
+	return (ret);
+}
+
+/* Compare two files by name */
+static int
+fts_compare_files(const FTSENT **left, const FTSENT **right)
+{
+	return (strcmp((*left)->fts_name, (*right)->fts_name));
```

Comment:
Do you want to make this case-insensitive too?
For what this is doing, you could also drop the user comparison function and just pass NULL as the last argument to `fts_open()` too.

Reply:
I wasn't exactly sure how the function was used by fts_read so didn't want to mess with it. You raise an excellent point, passing NULL means the compare function isn't required. That seems cleaner and works the same.